scrapy/spidermiddlewares/__init__.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+
+
+class BaseSpiderMiddleware(object):
+
+    def process_spider_input(self, response, spider):
+        # Called for each response that goes through the spider
+        # middleware and into the spider.
+
+        # Should return None or raise an exception.
```

Comment:
`pass` is basically saying `do nothing`. When a method returns nothing, it returns `None`. So this is mostly for documentation purposes to make it more clear that there is no implementation yet.